xchange-bittrex/src/main/java/org/knowm/xchange/bittrex/BittrexAdapters.java

```diff
@@ -219,14 +222,61 @@ public static Wallet adaptWallet(Collection<BittrexBalance> balances) {
     return Wallet.Builder.from(wallets).build();
   }
 
-  public static void adaptMetaData(List<BittrexSymbol> rawSymbols, ExchangeMetaData metaData) {
-    BittrexAdapters.adaptCurrencyPairs(rawSymbols)
-        .forEach(
-            currencyPair -> {
-              metaData.getCurrencyPairs().putIfAbsent(currencyPair, null);
-              metaData.getCurrencies().putIfAbsent(currencyPair.base, null);
-              metaData.getCurrencies().putIfAbsent(currencyPair.counter, null);
-            });
+  public static void adaptMetaData(List<BittrexSymbol> rawSymbols,
+                                   List<BittrexCurrency> bittrexCurrencies,
+                                   Map<CurrencyPair, Fee> dynamicTradingFees,
+                                   ExchangeMetaData metaData) {
+    List<CurrencyPair> currencyPairs = BittrexAdapters.adaptCurrencyPairs(rawSymbols);
+    for (CurrencyPair currencyPair : currencyPairs) {
+      CurrencyPairMetaData defaultCurrencyPairMetaData = metaData.getCurrencyPairs().get(currencyPair);
+      BigDecimal resultingFee = null;
+      // Prioritize dynamic fee
+      if( dynamicTradingFees != null ){
+        Fee fee = dynamicTradingFees.get(currencyPair);
+        if( fee != null){
+          resultingFee = fee.getMakerFee();
+        }
+      }else{
+        if (defaultCurrencyPairMetaData != null ){
+          resultingFee = defaultCurrencyPairMetaData.getTradingFee();
+        }
+      }
+
+      CurrencyPairMetaData newCurrencyPairMetaData;
+      if(defaultCurrencyPairMetaData != null){
```

Comment:
your formatting does not meat our conventions which indicates that you are building using your IDE tool chain and not maven - maven has an autoformatter configured

Reply:
fixed